modules/Auth/Auth.cpp

```diff
@@ -40,6 +47,10 @@ void Auth::ready() {
         evse_index++;
     }
 
+    this->p_main->publish_token_validation_status(types::authorization::TokenValidationStatus::Idle);
+
+    this->auth_handler->register_publish_token_validation_status_callback(
+        [this](types::authorization::TokenValidationStatus status) { this->p_main->publish_token_validation_status(status); });
```

Comment:
Is there any reason why this function couldn't just be a member function instead of a callback?  Instead of registering the callback, and calling its wrapper, the member function could be called directly

Reply:
This function is generated from the `auth.yaml` file and is a part of `authImpl` class.
The `Auth` class is the owner of the `AuthHandler` which contains the business logic of this module.
Since the `AuthHandler` class does not have any clue of the `Auth` class, it's only possible to pass the `authImplBase` pointer to the `AuthHandler`. That could be solved as a raw pointer (since it won't be owned by `AuthHandler` class). In that case we can than call the function directly. wdyt?

Reply:
Sorry, my fault - I understood something wrong here.  Still one might check, if the callback is set, before calling it.

Reply:
Should I try to do it within the whole Auth module?

Reply:
@wku12 no I don't think that this is your responsibility

Reply:
@a-w50 I understand your concern about the possibly callback not being set. 

However, this "pattern" of setting the callbacks in constructor and calling them from within `AuthHandler` object applies in the whole module. If someone forgot to add it there, the std::function will throw `std::bad_function_call` during execution of the code. This will cause the whole EVerest to crash which I personally think is better than not calling this callback and simply printing logs in that case.